### ISSUE TITLE
Frontmatter in partials displays in Middleman 4.1.8/4.1.9

### ISSUE THREAD

**Issue by jimothyGator** [middleman/middleman#1938]
If a partial contains YAML front matter, in Middleman 4.1.8 and 4.1.9 (but not 4.1.7 and earlier) will render that front matter verbatim.

Example:
### index.html.erb

```
Hello <%= partial 'world' %>
```
### _world.html.erb

```

---
foo: bar

---
world
```
### Expected Output

```
Hello world
```
### Actual Output

```
Hello ---
for: bar

---
world
```

**Comment by jimothyGator** [comment]
Added failing test case: #1939

**Comment by tdreyno** [comment]
Thanks for reporting and providing the case, but this is by design. Do not include front matter in partials.

**Comment by jimothyGator** [comment]
That's unfortunate to hear. We use Middleman to prototype sites which will be built on another platform. We use front matter to include meta data, from which we generate stubs for the production components. Prior to MM 4.1.8, this process worked.

**Comment by tdreyno** [comment]
Can you show me what this looks like in practice?

**Comment by jimothyGator** [comment]
It's a somewhat complex setup, and one that is particular to our use, but I'll do my best to explain it.

The final websites run in a CMS (Adobe Experience Manager [AEM], which is Java based), but before we build the components there, we build prototypes using Middleman, because this allows us to iterate much faster (particular because there is no compile and deploy cycle, as there is with AEM. This allows us to quickly create, debug, and refine our JavaScript and CSS/Sass; the same JavaScript and CSS files are shared between the prototype and the production site.

AEM components involve a number of files, much of it boilerplate. We've hooked in some Grunt tasks that generate these boilerplate files from the front matter we include in our Middleman partials. For each AEM component, we create a corresponding partial.

For example, we might define a header component in a partial named `_header.erb`, with the following front matter:

```
---
componentTitle: header
componentDescription: Our Site Header
componentGroup: Our Site
---
```

This would generate a variety of files for AEM: `Header.java`, `header.html`, `dialog.xml`, '.content.xml`, and the correct directory/package structure for them. The front matter values are used to create these files, which serve as starting points for the components. Developers then complete and customize the files. The code generation just provides a starting point, and automates a bit of the boilerplate and tedium away.

This is hardly a universal use case, to be sure, but others could probably find other uses for front matter, which was possible prior to 4.1.8.

**Comment by tdreyno** [comment]
Makes sense. Why does this component need to be a partial?

**Comment by jimothyGator** [comment]
Partials make sense because they are analogous to components: Both are building blocks for web pages. We intend to have (roughtly) a one-to-one relationship between our Middleman partials and AEM components. So the _header.erb partial serves as the prototype for the header component that will be built. Because that partial already existed, as we refined our process to automate more tasks, it made sense for us to stick the component properties in the partial. 

Of course, we could come up with alternatives (such as a separate file; e.g., header.yaml, located next to the partial for convenience). Since front matter in the partial worked in MM 3.x and 4.x until recently, though, we had developed a few components using this convention. Upon upgrading to MM 4.1.8, we then discovered that our front matter appeared on the compiled pages for our prototype.

It's true that it wouldn't be a huge effort for me to write a script that runs through our partials and extracts the front matter into separate files, and that's probably the approach we'll take if front matter support (or tolerance?) in the partials doesn't return. I just noticed the change and assumed it was inadvertent.
